server/src/test/java/org/apache/druid/segment/metadata/SegmentMetadataCacheCommon.java

```diff
@@ -0,0 +1,333 @@
+import org.apache.druid.query.aggregation.DoubleSumAggregatorFactory;
+import org.apache.druid.query.aggregation.LongSumAggregatorFactory;
+import org.apache.druid.query.aggregation.hyperloglog.HyperUniquesAggregatorFactory;
+import org.apache.druid.segment.IndexBuilder;
+import org.apache.druid.segment.QueryableIndex;
+import org.apache.druid.segment.incremental.IncrementalIndexSchema;
+import org.apache.druid.segment.writeout.OffHeapMemorySegmentWriteOutMediumFactory;
+import org.apache.druid.server.QueryLifecycleFactory;
+import org.apache.druid.server.QueryStackTests;
+import org.apache.druid.server.log.TestRequestLogger;
+import org.apache.druid.server.metrics.NoopServiceEmitter;
+import org.apache.druid.server.security.AuthConfig;
+import org.apache.druid.server.security.AuthTestUtils;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.partition.LinearShardSpec;
+import org.apache.druid.timeline.partition.NumberedShardSpec;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+public abstract class SegmentMetadataCacheCommon
+{
+  public static final String DATASOURCE1 = "foo";
+  public static final String DATASOURCE2 = "foo2";
+  public static final String DATASOURCE3 = "foo3";
+  public static final String SOME_DATASOURCE = "some_datasource";
+  public static final String TIMESTAMP_COLUMN = "t";
+  private static final InputRowSchema FOO_SCHEMA = new InputRowSchema(
+      new TimestampSpec(TIMESTAMP_COLUMN, "iso", null),
+      new DimensionsSpec(
+          DimensionsSpec.getDefaultSchemas(ImmutableList.of("dim1", "dim2", "dim3"))
+      ),
+      null
+  );
+
+  public final List<InputRow> ROWS1 = ImmutableList.of(
+      createRow(ImmutableMap.of("t", "2000-01-01", "m1", "1.0", "dim1", "")),
+      createRow(ImmutableMap.of("t", "2000-01-02", "m1", "2.0", "dim1", "10.1")),
+      createRow(ImmutableMap.of("t", "2000-01-03", "m1", "3.0", "dim1", "2"))
+  );
+
+  public final List<InputRow> ROWS2 = ImmutableList.of(
+      createRow(ImmutableMap.of("t", "2001-01-01", "m1", "4.0", "dim2", ImmutableList.of("a"))),
+      createRow(ImmutableMap.of("t", "2001-01-02", "m1", "5.0", "dim2", ImmutableList.of("abc"))),
+      createRow(ImmutableMap.of("t", "2001-01-03", "m1", "6.0"))
+  );
+
+  public QueryRunnerFactoryConglomerate conglomerate;
+  public Closer resourceCloser;
+  public QueryToolChestWarehouse queryToolChestWarehouse;
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  public QueryableIndex index1;
+  public QueryableIndex index2;
+
+  public QueryableIndex indexAuto1;
+  public QueryableIndex indexAuto2;
+
+  public DataSegment realtimeSegment1;
+  public DataSegment segment1;
+  public DataSegment segment2;
+  public DataSegment segment3;
+  public DataSegment segment4;
+  public DataSegment segment5;
+
+  static {
+    NullHandling.initializeForTests();
+  }
```

Comment:
nit: recommend extending `InitializedNullHandlingTest` since it also initializes `ExpressionProcessing` (the other static config), though since this probably isn't running any actual queries it probably isn't necessary..